File: ui/src/members/Member.ts
export class Member {
    id: string | undefined;
    first: string = '';
    last: string = '';
    email: string = '';
    password: string = '';
    cell: string = '';
    rating: number = 1200;
    isActive: boolean = false
    imageUrl: string = 'picture_placeholder.jpeg';

    constructor(initializer?: any) {
        if (!initializer) return;
        if (initializer.id) this.id = initializer.id;
        if (initializer.first) this.first = initializer.first;
        if (initializer.last) this.last = initializer.last;
        if (initializer.email) this.email = initializer.email;
        if (initializer.password) this.password = initializer.password;
        if (initializer.cell) this.cell = initializer.cell;
        if (initializer.rating) this.rating = initializer.rating;
        if (initializer.isActive) this.isActive = initializer.isActive;
        if (initializer.imageUrl) this.imageUrl = initializer.imageUrl;
    }
}

File: server/domain/model/member.go
package model

type Member struct {
	id       string
	first    string
	last     string
	email    string
	password string
	cell     string
	rating   int32
	isActive bool
	imageUrl string
}

func Create(
	id string,
	first string,
	last string,
	email string,
	password string,
	cell string,
	rating int32,
	isActive bool,
	imageUrl string,
) *Member {
	return &Member{
		id:       id,
		first:    first,
		last:     last,
		email:    email,
		password: password,
		cell:     cell,
		rating:   rating,
		isActive: isActive,
		imageUrl: imageUrl,
	}
}

func (cm Member) Id() string {
	return cm.id
}

func (cm Member) First() string {
	return cm.first
}

func (cm Member) Last() string {
	return cm.last
}

func (cm Member) Email() string {
	return cm.email
}

func (cm Member) Password() string {
	return cm.password
}

func (cm Member) Cell() string {
	return cm.cell
}

func (cm Member) Rating() int32 {
	return cm.rating
}

func (cm Member) IsActive() bool {
	return cm.isActive
}

func (cm Member) ImageUrl() string {
	return cm.imageUrl
}

func (cm Member) Empty() {}
